src/main/kotlin/Main.kt

```diff
@@ -2,98 +2,192 @@ import com.beust.klaxon.JsonArray
 import com.beust.klaxon.Klaxon
 import com.beust.klaxon.Parser.Companion.default
 import facts.RelationLoader
+import parsing.EngineConfiguration
+import parsing.Sorting
 import parsing.parseEngineConfigurations
 import proofs.addAllProofs
 import tests.Test
 import tests.testgeneration.addAllTests
 import java.io.File
+import java.nio.charset.Charset
+import java.nio.file.Paths
+import java.text.SimpleDateFormat
+import java.util.Date
 import java.util.concurrent.ConcurrentLinkedQueue
 import java.util.concurrent.atomic.AtomicInteger
 import kotlin.concurrent.thread
+import kotlin.io.path.absolutePathString
 import kotlin.math.roundToInt
 import kotlin.system.measureTimeMillis
 
 fun main() {
     val time = measureTimeMillis {
-        val tests = generateTests()
-        println("Found ${tests.size} tests")
-
-        val results = executeTests(tests)
-
+        val results = handleTests()
         saveResults(results.toList())
     }
 
     println()
     println("Done in ${time / 1000} seconds")
 }
 
-private fun generateTests(): Collection<Test> {
-    val allRelations = ProofSearcher().addAllProofs().findNewRelations(RelationLoader.relations)
-    //ProofLimiter(3).limit(allRelations)
-    return TestGenerator().addAllTests().generateTests(allRelations)
-}
-
 class ResultContext(val result: TestResult,
                     val engine: String,
                     val version: String)
 
-private fun executeTests(tests: Collection<Test>): Iterable<ResultContext> {
+private fun handleTests(): Iterable<ResultContext> {
```

Comment:
I think the name ``handleTests`` might be a bit too generic. THB I cant figure out a better name for it given its behavior. Maybe the function "handles" too much for it to only have a single purpose? From my understanding the function does:
- Iterates over all engine and sorts the total set of tests based on the engine configuration.
- Depending on the engine configuration it either omits running the tests and saves them, or it saves and runs them.
- At last it returns an iterable of all results.

Since ``executeTests`` takes an engine configuration, then saving and sorting the tests should maybe be moved into that function? (I am a bit unsure whether sorting should be moved into ``executeTests`` as the name implies that the tests passed to the function will be run). However, if saving the tests is moved into ``executeTests`` then the only behavior of ``handleTests`` would be job of delegating the tests to the engines and aggregating the results. In this case a name such as ``executeTestsOnAllEngines`` (or ``runTestsOnAllEngines``) might be saying more about its behavior and intent.